File: SortingAlgorithms/SelectionSort.java
public class SelectionSort {

	SelectionSort() {}

	public int[] sort (int[] arr) {

		
		for (int i =0; i< arr.length; i++) {
			int currMin = i;

			for (int j = i; j < arr.length; j++) {
			 	if (arr[j] < arr[currMin]) {
			 		currMin = j;
			 	}
			 }
			 int temp = arr[i];
			 arr[i] = arr[currMin];
			 arr[currMin] = temp;
		}

		return arr;
	}
	
	public static void main(String[] args) {
		int[] arr = {1, 6, 43, 3, 2};

		SelectionSort selectionSort = new SelectionSort();

		for (int ele: arr) {
			System.out.println(ele);
		}

		int[] newArr = selectionSort.sort(arr);
		System.out.println();
		
		for (int ele: newArr) {
			System.out.println(ele);
		}
	}
}

File: SortingAlgorithms/SelectionSort.cpp
#include <iostream>
#include <vector>
#include "/Algos/HelperClass/Vecs.h"

using namespace std;

class SelectionSort {

	public:
		void sort(vector<int> &vec) {

			size_t n = vec.size();

			for (size_t i = 0; i <n - 1; i++) {

				int smallest = i; 
				for (size_t j = i +1; j < n; j++) {
					if (vec[j] < vec[smallest]) {
						smallest = j;
					}	
				}

				int temp = vec[smallest];
				vec[smallest] = vec[i];
				vec[i] = temp;  
			}
		}
};

int main() {

	SelectionSort selectionSort;
	//Vecs vecs;
		
	vector<int> vec = {3, 32, 1, 34, 767};

	//vecs.printVector(vec);
	
	for (size_t i = 0; i < vec.size(); i++) {
		cout << vec[i] << " ";
	}

	cout << endl;

	selectionSort.sort(vec);

	
	for (size_t i = 0; i < vec.size(); i++) {
		cout << vec[i] << " ";
	}
	
	cout << endl;
}
